### PR TITLE
feat: Same incidents in the past

### PR DESCRIPTION
**Pull request by Matvey-Kuk**
Linking incidents.

Closes https://github.com/keephq/keep/issues/2137

### CODE REVIEW

File: keep/api/core/db.py [hunk 1]
```diff
@@ -2842,6 +2843,30 @@ def get_incident_alerts_by_incident_id(
...
+def get_future_incidents_by_incident_id(
+    incident_id: str,
+    limit: Optional[int] = None,
+    offset: Optional[int] = None,
+) -> (List[Incident], int):
+    with Session(engine) as session:
+        query = (
+            session.query(
+                Incident,
+
```
Comment (VladimirFilonov) [inline keep/api/core/db.py:2856]: Useless empty line

File: keep/api/models/db/migrations/versions/2024-10-08-10-47_bf756df80e9d.py
Comment (VladimirFilonov) [inline keep/api/models/db/migrations/versions/2024-10-08-10-47_bf756df80e9d.py:1]: Why separate migration?

**Review by VladimirFilonov — CHANGES_REQUESTED** [review]

File: keep/api/models/db/migrations/versions/2024-07-25-17-13_67f1efb93c99.py [hunk 2]
```diff
@@ -29,15 +31,18 @@
...
-incident_table = sa.Table(
-    'incident',
-    migration_metadata,
-    sa.Column('id', UUID(as_uuid=False), primary_key=True),
-    sa.Column('alerts_count', sa.Integer, default=0),
-    sa.Column('affected_services', sa.JSON, default_factory=list),
-    sa.Column('sources', sa.JSON, default_factory=list)
-)
+# The following code will shoow SA warning about dialect, so we suppress it.
+with warnings.catch_warnings():
```
Comment (Matvey-Kuk) [inline keep/api/models/db/migrations/versions/2024-07-25-17-13_67f1efb93c99.py:35]: Here I fixed the old migration, it's been polluting logs with warnings.

**Review by VladimirFilonov — APPROVED** [review]
LGTM

**Pull request merged**